spyder/plugins/variableexplorer/widgets/namespacebrowser.py

```diff
@@ -118,21 +149,6 @@ def set_shellwidget(self, shellwidget):
         self.shellwidget = shellwidget
         shellwidget.set_namespacebrowser(self)
 
-    def set_text_finder(self, text_finder):
-        """Bind NamespaceBrowsersFinder to namespace browser."""
-        self.text_finder = text_finder
-        if self.finder_is_visible:
-            self.text_finder.setText(self.last_find)
-        self.editor.finder = text_finder
-
-        return self.finder_is_visible
-
-    def save_finder_state(self, last_find, finder_visibility):
-        """Save last finder/search text input and finder visibility."""
-        if last_find and finder_visibility:
-            self.last_find = last_find
-        self.finder_is_visible = finder_visibility

```

Comment:
This is a loss of functionality, i.e. in the current implementation we're saving the last text that was introduced by the user in `text_finder` and we restore for the next search. That's very useful to switch back and forth between all variables and the ones you want to filter when running code several times.

So, please try to restore it in `FinderWidget`.

Reply:
Done